solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -252,6 +253,11 @@ public CoreLoadFailure(CoreDescriptor cd, Exception loadFailure) {
           getZkController().getOverseer() != null &&
           !getZkController().getOverseer().isClosed(),
       (r) -> this.runAsync(r));
+
+  private final ClusterEventProducerFactory clusterEventProducerFactory = new ClusterEventProducerFactory(this);
```

Comment:
Does it have to be a field? It can be a local variable as well

Reply:
+1.